test/index.test.js

```diff
@@ -1,36 +1,632 @@
-'use strict';
+'use strict'
 
-const { test } = require('tap');
-const nock = require('nock');
-const { run } = require('../src/run');
+const { test } = require('tap')
+const nock = require('nock')
+const { run } = require('../src/run')
 
-const TOKEN = 'jifeo2903u089jf3920';
-const TEAM_ID = 'i992-j9f23j09-j092j0';
+// Import fixtures
+const push_payload = require('./fixtures/push')
+const pull_request_payload = require('./fixtures/pull_request')
+const deployment_payload = require('./fixtures/deployment')
+const deployment_status_payload = require('./fixtures/deployment_status')
+const status_payload = require('./fixtures/status')
+const package_payload = require('./fixtures/package')
+const release_payload = require('./fixtures/release')
+const issue_payload = require('./fixtures/issue')
+
+const TOKEN = 'jifeo2903u089jf3920'
+const TEAM_ID = 'i992-j9f23j09-j092j0'
 
 const setInput = (name, value) => {
-  process.env[`INPUT_${name.replace(/ /g, '_').toUpperCase()}`]=value;
+  process.env[`INPUT_${name.replace(/ /g, '_').toUpperCase()}`] = value
 }
 
 const clearInput = (name) => {
-  delete process.env[`INPUT_${name.replace(/ /g, '_').toUpperCase()}`];
+  delete process.env[`INPUT_${name.replace(/ /g, '_').toUpperCase()}`]
 }
 
 test('Push event with only required fields has additional data added', async ({ is }) => {
-  setInput('token', TOKEN);
-  setInput('team_id', TEAM_ID);
-  setInput('event_name', 'deployment');
-  setInput('event_action', 'succeeded');
+  setInput('token', TOKEN)
+  setInput('team_id', TEAM_ID)
+  setInput('event_name', 'deployment')
+  setInput('event_action', 'succeeded')
 
   const context = {
     eventName: 'push',
-    payload: {
-      ref: 'refs/heads/main',
-      after: '12345',
-      repository: {
-        full_name: 'org/test'
-      }
-    }
-  };
+    payload: push_payload
+  }
+
+  const req = nock('https://events.cto.ai')
+    .post('/', {
+      team_id: TEAM_ID,
+      event_name: 'deployment',
+      event_action: 'succeeded',
+      environment: null,
+      image: null,
+      branch: 'simple-tag',
+      commit: '6113728f27ae82c7b1a177c8d03f9e96e0adf246',
+      repo: 'Codertocat/Hello-World'
+    })
+    .reply(200, { message: 'event written', data: {} })
+
+  await run(context)
+  is(req.isDone(), true)
```

Comment:
test the response output not the nock state, that way we can confidently run these as integration tests using `NOCK_OFF=true npm test` - might be worth adding an `integration` package.json script that does just that